src/core/utils/chainProviders.ts

```diff
@@ -1,100 +1,110 @@
 import { Network, ethers } from 'ethers'
+import { Config } from '../../config'
 import { Multicall__factory } from '../../contracts'
 import { Chain } from '../constants/chains'
 import { MULTICALL_ADDRESS } from '../constants/MULTICALL_ADDRESS'
 import { CustomMulticallJsonRpcProvider } from './CustomMulticallJsonRpcProvider'
 import { logger } from './logger'
 import { MulticallQueue } from './multicall'
 
-const provider = ({
-  url,
-  chainId,
-  enableMulticallQueue,
-}: {
-  url: string | undefined
-  chainId: Chain
-  enableMulticallQueue: boolean
-}) => {
-  if (!url) return undefined
+export class ChainProvider {
+  private config: Config
+  providers: Record<Chain, ethers.JsonRpcProvider>
 
-  if (!enableMulticallQueue) {
-    logger.debug({ chainId }, `Using standard provider`)
-    return new ethers.JsonRpcProvider(url, chainId, {
-      staticNetwork: Network.from(chainId),
-    })
+  constructor(config: Config) {
+    this.config = config
```

Comment:
I don't think you need to store the config. Just pass it to `initializeProviders` so it can be used there.